apps/ledger-live-mobile/src/components/RequiresBLE/hooks/useAndroidBluetoothPermissions.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  Permission,
+  PermissionsAndroid,
+  Platform,
+  AppState,
+} from "react-native";
+import { useRef, useCallback, useState, useEffect } from "react";
+
+const { PERMISSIONS, RESULTS } = PermissionsAndroid;
+
+// No check is done on iOS here
+export const bluetoothPermissions: Permission[] =
+  Platform.OS === "android"
+    ? Platform.Version < 31
+      ? []
+      : [PERMISSIONS.BLUETOOTH_SCAN, PERMISSIONS.BLUETOOTH_CONNECT]
+    : [];
+
+/**
+ * Checks if the app has the required permissions to use Bluetooth.
+ *
+ * On Android 11 and above, the app needs to have the BLUETOOTH_SCAN and BLUETOOTH_CONNECT permissions.
+ * On iOS, no check is done and returns true.
+ */
+async function checkBluetoothPermissions() {
+  if (bluetoothPermissions.length === 0) return true;
+
+  return Promise.all(
+    bluetoothPermissions.map(permission =>
+      PermissionsAndroid.check(permission),
+    ),
+  ).then(results => results.every(res => res));
+}
+
+type RequestMultipleResult = {
+  allGranted: boolean;
+  generalStatus: typeof PermissionsAndroid.RESULTS[number];
+};
+
+/**
+ * Requests the required permissions to use Bluetooth.
+ *
+ * On Android 11 and above, the app needs to have the BLUETOOTH_SCAN and BLUETOOTH_CONNECT permissions.
+ * On iOS, no request is done and returns true.
+ */
+async function requestBluetoothPermissions(): Promise<RequestMultipleResult> {
+  if (bluetoothPermissions.length === 0)
+    return { allGranted: true, generalStatus: RESULTS.GRANTED };
+
+  return PermissionsAndroid.requestMultiple(bluetoothPermissions).then(res => {
+    const permissionStatuses = Object.values(res);
+    const allGranted = permissionStatuses.every(
+      status => status === RESULTS.GRANTED,
+    );
+
+    /** https://developer.android.com/about/versions/11/privacy/permissions#dialog-visibility */
+    const someNeverAskAgain = permissionStatuses.some(
+      status => status === RESULTS.NEVER_ASK_AGAIN,
+    );
+
+    return {
+      allGranted,
+      generalStatus: allGranted
+        ? RESULTS.GRANTED
+        : someNeverAskAgain
+        ? RESULTS.NEVER_ASK_AGAIN
+        : RESULTS.DENIED,
+    };
+  });
+}
+
+/**
+ * Hook to check and request the BLE permission on Android.
```

Comment:
Maybe here document what happens at mount (automatic check and request of the permission)